src/chat.rs

```diff
@@ -2492,8 +2492,9 @@ async fn prepare_msg_blob(context: &Context, msg: &mut Message) -> Result<()> {
         if msg.viewtype == Viewtype::Image
             || maybe_sticker && !msg.param.exists(Param::ForceSticker)
         {
-            blob.recode_to_image_size(context, &mut maybe_sticker)
-                .await?;
+            if let Err(err) = blob.recode_to_image_size(context, &mut maybe_sticker).await {
+                warn!(context, "Failed to recode image: {err:#}.");
```

Comment:
So, EXIF won't be removed, but i don't know how critical it is

Reply:
in general, unrelated to this special issue, i think, EXIF should not rule everything. doing best effort on EXIF removal on the "happy path" of taking a photo and sending it seems to be the gist.

if the user has edited an image (or attempted to do so), it will be an endless struggle to protect the users from themselves - the user could have saved the image also in a EXIF-preserving-format not understood by us. or the user can send an image as "File" would also preserve EXIF and is only one tap away.

Reply:
Overall i agree, but looks like we can use here `image::load_from_memory()` instead of `image::open()` or even `load_from_memory_with_format()` first, assuming that the extension is correct.

Reply:
I'm going to try the `image::load_from_memory()` approach, it seems to be not much more difficult than just using `image::open()`